### PR TITLE
fix(effects): resubscribe every time an error occurs

### PR DESCRIPTION
**Pull request by timdeschryver**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngrx/platform/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Documentation has been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Closes #

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

## Other information

Seems like with the original PR we forgot to modify the spec for the usage with `createEffect`. This PR uses the same implementation as the original PR.

History:

- #2026 - original PR with the fix
- #2037  - revert caused by inconsistant CI

### CODE REVIEW

File: modules/effects/src/effects_resolver.ts [hunk 1]
```diff
@@ -24,14 +24,19 @@ export function mergeEffects(
...
+      const resubscribeInCaseOfError = (
+        observable$: Observable<any>
+      ): Observable<any> =>
+        observable$.pipe(
+          catchError(error => {
+            if (errorHandler) errorHandler.handleError(error);
+            // Return observable that produces this particular effect
+            return resubscribeInCaseOfError(observable$);
+          })
+        );
```
Comment (alex-okrushko) [inline modules/effects/src/effects_resolver.ts]: Can you please make it as a package-private function declaration and put it outside of mergeEffects?

File: modules/effects/src/effects_resolver.ts [hunk 2]
```diff
@@ -24,14 +24,19 @@ export function mergeEffects(
           ? sourceInstance[propertyName]()
           : sourceInstance[propertyName];
 
+      const resubscribeInCaseOfError = (
+        observable$: Observable<any>
```
Comment (alex-okrushko) [inline modules/effects/src/effects_resolver.ts]: both `any` should be replaced with `T` to make sure that both Observables are of the same kind.

File: modules/effects/src/effects_resolver.ts [hunk 3]
```diff
@@ -61,3 +50,19 @@ export function mergeEffects(
 
   return merge(...observables$);
 }
+
+/**
+ * @internal
+ */
+export function resubscribeInCaseOfError<T extends Action = Action>(
```
Comment (alex-okrushko) [inline modules/effects/src/effects_resolver.ts]: No need to export it, right?

Comment (alex-okrushko) [inline modules/effects/src/effects_resolver.ts]: Is `= Action` needed? I think just extending Action is fine.

Comment (timdeschryver) [reply modules/effects/src/effects_resolver.ts]: Right, it also doesn't need the `@internal` comment

Comment (timdeschryver) [reply modules/effects/src/effects_resolver.ts]: No, it isn't needed. Thanks for the review 👍

**Pull request merged**